fastcrawler/test_utils.py

```diff
@@ -1,62 +1,73 @@
-from fastcrawler.engine import AioHttpEngine
+from enum import StrEnum
 
+from fastcrawler.engine.contracts import (
+    ProxySetting,
+    RequestCycle,
+    SetCookieParam,
+    Response,
+    Url,
+    Request,
+    EngineProto,
+)
 
-class TestServer:
-    def __init__(self):
-        self.responses = {
-            "GET": {"message": "GET request received"},
-            "POST": {"message": "POST request received"},
-            "PUT": {"message": "PUT request received"},
-            "DELETE": {"message": "DELETE request received"},
-        }
 
-    def get_response(self, method, data=None):
-        response = self.responses.get(method)
-        if data:
-            response["data"] = data
-        return response
+class Method(StrEnum):
+    GET = "GET"
+    POST = "POST"
+    PUT = "PUT"
+    DELETE = "DELETE"
 
 
-class TestClient(AioHttpEngine):
-    def __init__(self, server=TestServer(), headers=None, proxy=None):
-        self.server = server
+class MockEngine:
+    def __init__(
+        self,
+        cookies: list[SetCookieParam] | None = None,
+        headers: dict | None = None,
+        user_agent: str | None = None,
+        proxy: ProxySetting | None = None,
+        connection_limit: int | None = None,
+    ):
+        super().__init__(cookies, headers, user_agent, proxy, connection_limit)
```

Comment:
what is super()??? you don't inheritance from anything!

Reply:
will removed